new product overview page (#2475)
pulumi/pulumi-hugo: themes/default/content/product/_index.md

    and tame cloud complexity -- something we call Cloud Engineering.

overview:
    title: Universal Infrastructure as Code for Engineers
    description: |
        Ship infrastructure faster with software engineering and an API-first approach. Use Pulumi’s open source SDK with your favorite languages and software ecosystems to deploy on any cloud, and scale infrastructure as code across your team securely with enterprise-grade SaaS.

key_features:
    title: Key features
    items:
        - title: "Author in any language, deploy to any cloud"
          sub_title: "Pulumi Infrastructure as Code Engine"
          description: 

>             Define infrastructure as code (IaC) in TypeScript/JavaScript, Python, Go, C#, Java, and YAML using your IDE and test frameworks for a fast inner dev loop. Deploy to [100+ providers](/registry/) like AWS, Azure, Google Cloud, and Kubernetes.

Reviewer: global comment: these text blocks appear quite wide on my screen and might benefit from being compressed a little

>           image: "/images/productnew/pulumi-iac-code.png"

Author: this is the original. maybe I have the image expanded a bit too big.

          button:
            text: "Learn more about Pulumi SDK"
            link: "/docs/reference/pulumi-sdk/"
          features:
              - title: IaC for every engineer
                description: |
                   Author and share infrastructure code in standard languages that anyone can easily use.
              - title: Author IaC efficiently
                description: |
                    Use your IDE with interactive documentation, type checking, and statement completion.
              - title: Write Policy as Code
                description: |
                    Write Policy as Code in programming languages to enforce best practices with [Crossguard](/crossguard/).

        - title: "Create infrastructure automation workflows"
          sub_title: "Pulumi Automation API"
          description: |
            Create workflows that coordinate provisioning, previewing, refreshing, and destroying cloud resources by using the Pulumi engine as a library in your application code.

>           image: "/images/product/automation-api.png"

Reviewer: let me get you an updated version of this

          button:
            text: "Learn more about Automation API"
            link: "/docs/reference/pulumi-sdk/"
          features:
              - title: 10x productivity
                description: |
                   Engineers can manage 10x more cloud resources compared to traditional CLI tools.
              - title: Build custom CLIs
                description: |
                    Create CLIs that make it easy for your developers to provision prebuilt architectures.
              - title: Power up your SaaS
                description: |
                    Enable your cloud service to dynamically provision and manage cloud resources at scale.

        - title: "Build infrastructure faster with reusable components"
          sub_title: "Pulumi Packages"
          description: |
            Build and reuse higher-level abstractions for cloud architectures with multi-language Pulumi Packages. Distribute the packages through repositories or package managers so your team members can reuse them.

>           image: "/images/product/pulumi-packages.png"

Reviewer: instead of an image, can you put the code example box? it's from here: https://www.pulumi.com/registry/packages/eks/

Author: maybe... let me try and see. the way I have it abstracted, it assumes an image for this section. But will see if I can get the code box to work,.

Author: i added in the code box instead.

          button:
            text: "Learn more about CI/CD Integrations"
            link: "/docs/reference/pulumi-sdk/"
          features:
              - title: Version and review
                description: |
                    Manage infrastructure code in Git and approve changes through pull requests.
              - title: Shift left
                description: |
                    Run fast, local [unit tests and integration tests](/docs/guides/testing/) against ephemeral infrastructure.
              - title: Continuous delivery
                description: |
                    [Integrate your CI/CD provider](/docs/guides/continuous-delivery/) with Pulumi or use GitOps to manage Kubernetes clusters.
        
        - title: "Scale and secure infrastructure as code for teams"
          sub_title: "Pulumi Service"
          description: |
             Store infrastructure state & secrets, empower teams to ship infrastructure collaboratively, and manage security and governance using the Pulumi Service. The fastest and easiest way to use Pulumi at scale. 
          image: "/images/productnew/pulumi-service.png"
          button:

>             text: "Learn more about Pulumi SDK"

Reviewer: ```suggestion
            text: "Learn more about the Pulumi Service"
```

    and tame cloud complexity -- something we call Cloud Engineering.

overview:

>     title: Infrastructure as Code for Engineers

Reviewer: there's a lot of whitespace above the "Infrastructure as code for Engineers" and below the "Get Started" button. do you think it would look better reduced?

Author: agreed. I reduced it a bit.

    description: |

>         Ship infrastructure faster with software engineering. Use Pulumi’s open source SDK in your favorite languages to provision infrastructure on any cloud, and securely increase infrastructure as code adoption across teams with enterprise-grade SaaS.

Reviewer: ```suggestion
        Ship infrastructure faster with software engineering. Use Pulumi’s open source SDK in your favorite languages to provision infrastructure on any cloud, and securely increase infrastructure as code adoption across teams with Pulumi's enterprise-grade SaaS.
```

            link: "/docs/reference/pulumi-sdk/"
          features:
              - title: IaC for every engineer
                description: |
                   Author and share infrastructure code in standard languages that anyone can easily use.
              - title: Author IaC faster
                description: |
                    Use your IDE with interactive documentation, type checking, and statement completion.
              - title: Write Policy as Code
                description: |
                    Write Policy as Code in programming languages to enforce best practices with [Crossguard](/crossguard/).

        - title: "Create infrastructure automation workflows"
          sub_title: "Pulumi Automation API"
          description: |
            Create workflows that coordinate provisioning, previewing, refreshing, and destroying cloud resources by using the Pulumi engine as a library in your application code.
          image: "/images/product/automation-api.png"
          button:
            text: "Learn more about Automation API"
            link: "/automation/"
          features:
              - title: 10x productivity
                description: |

>                    Engineers can manage 10x more cloud resources compared to traditional CLI tools.

Reviewer: ```suggestion
                   Engineers can manage 10x more cloud resources using Automation API compared to traditional CLI tools.
```

>               - title: Build custom CLIs

Reviewer: ```suggestion
              - title: Create custom CLIs
```

                description: |

>                     Create CLIs that make it easy for your developers to provision prebuilt architectures.

Reviewer: ```suggestion
                    Build atop Pulumi to create CLIs that make it easy for end-users to provision prebuilt cloud architectures.
```

              - title: Power up your SaaS
                description: |

>                     Enable your cloud service to dynamically provision and manage cloud resources at scale.

Reviewer: ```suggestion
                    Enable your services and APIs to dynamically provision and manage cloud resources at scale.
```

                        Pulumi.run(App::stack);
                    }

                    private static void stack(Context ctx) {
                    final var cluster = new Cluster("eks-cluster");
                    ctx.export("kubeconfig", cluster.kubeconfig());
                  }
                }
            - title: Pulumi.yaml
              language: yaml
              code: |
                resources:
                  eks-cluster:
                    type: eks:Cluster
                outputs:
                  kubeconfig: ${cluster.kubeconfig}
          button:
            text: "Learn more about Pulumi Packages"
            link: "/product/packages/"
          features:
              - title: Native cloud providers
                description: |
                    Full API coverage for AWS, Azure, Google Cloud, and Kubernetes with same-day updates.
              - title: Crosswalk for AWS
                description: |
                    Adopt well-architected best practices for your infrastructure easily with the [Crosswalk library](/docs/guides/crosswalk/aws/).
              - title: Cloud Native support
                description: |

>                     Use a single workflow to manage cloud native infrastructure and [Kubernetes](/kubernetes/).

Reviewer: ```suggestion
                    Use a single workflow and programming languages to manage both [Kubernetes](/kubernetes/) resources and infrastructure.
```

>                     Use a single workflow and programming languages to manage both [Kubernetes](/kubernetes/) resources and infrastructure.

Reviewer: ```suggestion
                    Use a single workflow to manage both [Kubernetes](/kubernetes/) resources and infrastructure.
```

        
        - title: "Deliver infrastructure through software delivery pipelines"
          sub_title: "CI/CD Integrations"
          description: |
            Version, review, test, and deploy infrastructure code through the same tools and processes used for your application code.

>           image: "/images/product/pulumi-cicd.png"

Reviewer: the whitespace in the graphic is causing it to look like there is more whitespace between image & text here than in the other sections

Author: not sure how to get around this as this is a template that is abstracted to repeat itself for all the key features. I can maybe photoshop the image a bit to crop out the whitespace border around the logos.

Author: @gthuang I photoshopped out some of the surrounding whitespace in the image. let me know what you think.

          button:
            text: "Learn more about CI/CD Integrations"
            link: "/docs/guides/continuous-delivery/"
          features:
              - title: Version and review
                description: |
                    Manage infrastructure code in Git and approve changes through pull requests.
              - title: Shift left
                description: |
                    Get rapid feedback on your code with fast [unit tests](/docs/guides/testing/), and run [integration tests](/docs/guides/testing/) against ephemeral infrastructure.
              - title: Continuous delivery
                description: |

>                     [Integrate your CI/CD provider](/docs/guides/continuous-delivery/) with Pulumi or use GitOps to manage Kubernetes clusters.

Reviewer: ```suggestion
                    [Integrate your CI/CD provider](/docs/guides/continuous-delivery/) with Pulumi or use GitOps to [manage Kubernetes clusters](/docs/guides/continuous-delivery/pulumi-kubernetes-operator/).
```

>         - title: "Scale and secure infrastructure as code for teams"

Reviewer: the screenshot is not showing up

Author: ahh sorry, i forgot to commit the file. should be good now.
